packages/textlint-tester/src/textlint-tester.ts

```diff
@@ -71,47 +119,109 @@ export class TextLintTester {
         }
     }
 
-    testValidPattern(ruleName: string, rule: TextlintRuleCreator, valid: TesterValid) {
+    testValidPattern(name: string, param: TextlintRuleCreator | TestConfig, valid: TesterValid) {
         const text = typeof valid === "object" ? valid.text : valid;
         const inputPath = typeof valid === "object" ? valid.inputPath : undefined;
-        const options = (typeof valid === "object" && valid.options) || {};
         const ext = typeof valid === "object" && valid.ext !== undefined ? valid.ext : ".md";
         const textlint = new TextLintCore();
-        textlint.setupRules(
-            {
-                [ruleName]: rule
-            },
-            {
-                [ruleName]: options
+        if (isTestConfig(param)) {
+            const rules: any = {};
+            const rulesOptions: any = {};
+            param.rules.forEach(rule => {
+                const ruleName = rule.ruleId;
+                const ruleObject = rule.rule;
+                const ruleOptions = rule.options || {};
+                rules[ruleName] = ruleObject;
+                rulesOptions[ruleName] = ruleOptions;
+            });
+            textlint.setupRules(rules, rulesOptions);
+            if (param.plugins !== undefined) {
+                param.plugins.forEach(plugin => {
+                    const pluginName = plugin.pluginId;
+                    const pluginObject = plugin.plugin;
+                    const pluginOptions = plugin.options || {};
+                    textlint.setupPlugins(
+                        {
+                            [pluginName]: pluginObject
+                        },
+                        {
+                            [pluginName]: pluginOptions
+                        }
+                    );
+                });
```

Comment:
we can apply same approch to setupPlugins.
Create a wrap `createPluginXXX` and use it.

Reply:
Created `createTestPluginSet` function.